src/js/components/SidebarComponent.jsx

```diff
@@ -0,0 +1,106 @@
+var Link = require("react-router").Link;
+var React = require("react/addons");
+
+var AppListLabelsFilterComponent =
+  require("../components/AppListLabelsFilterComponent");
+var AppListStatusFilterComponent =
+  require("../components/AppListStatusFilterComponent");
+
+var SidebarComponent = React.createClass({
+  displayName: "SidebarComponent",
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    groupId: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filters: {
+        filterLabels: [],
+        filterStatus: []
+      }
+    };
+  },
+
+  updateFilterLabels: function (filterLabels) {
+    var state = this.state;
+
+    this.setState(Object.assign(state.filters, {
```

Comment:
No, I don't modify ```this.state.filterLabels```, because there isn't one anymore.
There is only a ```this.state.filters``` now, which is an object with an unchanging reference.
```filterLabels``` now lives in ```this.state.filters``` as ```this.state.filters.filterLabels```.
You see where my question comes from?
```this.state.filters``` never changes it's reference, so the state is actually not modified directly.